src/components/ui/Radio/index.tsx

```diff
@@ -40,17 +47,36 @@ export const Radio: FC<RadioProps> = props => {
         value={value}
         ref={radioNode}
         data-testid="hidden-radio"
+        aria-label={label}
         {...rest}
       />
-      <StyledRadio
-        checked={checked}
-        className="ch-radio"
-        onClick={handleChange}
-        data-testid="styled-radio"
-      />
-      <Label htmlFor={labelId} className="ch-radio-label">
-        {label}
-      </Label>
+
+      {icon ? (
+        <>
+          <StyledRadioIcon
+            checked={checked}
+            className="ch-radio"
+            onClick={handleChange}
+            data-testid="styled-radio-icon"
+          >
+            <span className="ch-icon">
+              {icon}
+            </span>
+          </StyledRadioIcon>
```

Comment:
LGTM - has the accessibility of this been verified? I'm assuming it's fine since you're adding the aria-label

Reply:
Verified via VoiceOver on the Mac and tabbing through and confirmed that the aria-label is read.